syntax = "proto3";


package torch2trt;


enum DataTypeMsg {
  kFloat = 0;
  kHalf = 1;
  kInt8 = 2;
  kInt32 = 3;
}

message TensorShapeMsg {
  repeated int64 size = 1; // does not include batch
}

// BEGIN TORCH METHODS

message InterpolateMsg {
  repeated int64 size = 1;
  string mode = 2;
  bool align_corners = 3;
}

// END TORCH METHODS

message TorchPluginMsg {
  DataTypeMsg dtype = 1;
  repeated TensorShapeMsg input_shapes = 2;
  repeated TensorShapeMsg output_shapes = 3;
  
  oneof method {
    InterpolateMsg interpolate = 4;
    // ADD TORCH METHOD HERE
  }
}